patches/minecraft/net/minecraft/entity/EntityLivingBase.java.patch

```diff
@@ -141,7 +173,33 @@
              }
  
              this.field_70170_p.func_72960_a(this, (byte)3);
-@@ -1253,15 +1273,7 @@
+@@ -1195,19 +1222,22 @@
+ 
+     public void func_70653_a(Entity p_70653_1_, float p_70653_2_, double p_70653_3_, double p_70653_5_)
+     {
++        LivingKnockBackEvent event = net.minecraftforge.common.ForgeHooks.onLivingKnockBack(this, p_70653_1_, p_70653_2_, p_70653_3_, p_70653_5_);
++        if(event.isCanceled()) return;
++        p_70653_2_ = event.getStrength(); p_70653_3_ = event.getRatioX(); p_70653_5_ = event.getRatioZ();
+         if (this.field_70146_Z.nextDouble() >= this.func_110148_a(SharedMonsterAttributes.field_111266_c).func_111126_e())
+         {
+             this.field_70160_al = true;
+             float f = MathHelper.func_76133_a(p_70653_3_ * p_70653_3_ + p_70653_5_ * p_70653_5_);
+             this.field_70159_w /= 2.0D;
+             this.field_70179_y /= 2.0D;
+-            this.field_70159_w -= p_70653_3_ / (double)f * (double)p_70653_2_;
+-            this.field_70179_y -= p_70653_5_ / (double)f * (double)p_70653_2_;
++            this.field_70159_w -= p_70653_3_ / (double)f * (double) p_70653_2_;
++            this.field_70179_y -= p_70653_5_ / (double)f * (double) p_70653_2_;
+ 
+             if (this.field_70122_E)
+             {
+                 this.field_70181_x /= 2.0D;
+-                this.field_70181_x += (double)p_70653_2_;
++                this.field_70181_x += (double) p_70653_2_;
```

Comment:
You have added a space here `(double) p_70653_2_` which causes additional lines in the patch file that do precisely nothing.